### PR TITLE
Add codeowners file

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**
Adds a codeowners file to enable auto review assignment.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Lines starting with '#' are comments.
+# Each line is a file pattern followed by one or more owners.
+
+*                    @protolambda @norswap @trianglesphere @maurelian @karlfloersch @ben-chain @cfromknecht @mslipper @tynes
+/l2os/               @protolambda @norswap @trianglesphere @cfromknecht @mslipper @tynes
+/opnode/             @protolambda @norswap @trianglesphere @cfromknecht @mslipper @tynes
+/packages/contracts/ @protolambda @norswap @maurelian @karlfloersch @smartcontracts
```
Comment (maurelian) [inline .github/CODEOWNERS:7]: This will automatically assign reviews accordingly right? Seems like it'll be super noisy.

Comment (protolambda) [reply .github/CODEOWNERS:7]: 5+ reviewers per PR by default is indeed quite noisy

Comment (trianglesphere) [reply .github/CODEOWNERS:7]: We need to make sure that it is configured to select 1 or 2 reviewers (can be done somewhere in github), but yes, the intention is to automatically select reviewers.

Comment (trianglesphere) [reply .github/CODEOWNERS:7]: https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team

Comment (maurelian) [reply .github/CODEOWNERS:7]: IMO we should just choose one or two ppl who can actually _own_ each area of the repo. Either those people can review the pr, or will take responsibility for finding someone who can.

**Comment by mslipper** [comment]
I'm very much in favor of having codeowners. However, I'll echo @protolambda's feedback that 5+ owners per area of the repo will produce too much noise. We also want the codeowners to actually perform the review, and having lots of code owners makes it really easy to ignore the review request because "someone else will do it." I think reducing it to 2 people per area is the best place for us to start.

**Pull request closed without merging**